caravel/views.py

```diff
@@ -886,51 +887,82 @@ def explore(self, datasource_type, datasource_id):
         datasources = sorted(datasources, key=lambda ds: ds.full_name)
         datasource = [ds for ds in datasources if int(datasource_id) == ds.id]
         datasource = datasource[0] if datasource else None
-        slice_id = request.args.get("slice_id")
-        slc = None
 
-        if slice_id:
-            slc = (
-                db.session.query(models.Slice)
-                .filter_by(id=slice_id)
-                .first()
-            )
         if not datasource:
             flash(__("The datasource seems to have been deleted"), "alert")
             return redirect(error_redirect)
 
-        slice_add_perm = self.can_access('can_add', 'SliceModelView')
-        slice_edit_perm = check_ownership(slc, raise_if_false=False)
-        slice_download_perm = self.can_access('can_download', 'SliceModelView')
-
         all_datasource_access = self.can_access(
             'all_datasource_access', 'all_datasource_access')
+
         datasource_access = self.can_access(
             'datasource_access', datasource.perm)
+
         if not (all_datasource_access or datasource_access):
             flash(__("You don't seem to have access to this datasource"),
                   "danger")
             return redirect(error_redirect)
 
-        action = request.args.get('action')
+        # handle slc / viz obj
+        slice_id = slice_id if slice_id != None else request.args.get("slice_id")
```

Comment:
does this work for `slice_id = 0` because slice_id is actually a string? thinking too much in JS.